x/concentrated-liquidity/swaps.go

```diff
@@ -203,16 +205,18 @@ func (k Keeper) CalcInAmtGivenOut(
 // calcOutAmtGivenIn calculates tokens to be swapped out given the provided amount and fee deducted. It also returns
 // what the updated tick, liquidity, and currentSqrtPrice for the pool would be after this swap.
 // Note this method is non-mutative, so the values returned by CalcOutAmtGivenIn do not get stored
+// Instead, we return writeCtx function so that the caller of this method can decide to write the cached ctx to store or not.
```

Comment:
Do you mind adding spec for `calcInAmtGivenOut` as a drive by change in this PR please?

Just noticed it's missing